airbyte-integrations/connectors/source-mixpanel/source_mixpanel/source.py

```diff
@@ -79,15 +79,26 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
         try:
             config = self._validate_and_transform(config)
             auth = self.get_authenticator(config)
-            FunnelsList.max_retries = 0
-            funnels = FunnelsList(authenticator=auth, **config)
-            funnels.reqs_per_hour_limit = 0
-            next(read_full_refresh(funnels), None)
-        except requests.HTTPError as e:
-            return False, e.response.json()["error"]
         except Exception as e:
             return False, e
 
+        # https://github.com/airbytehq/airbyte/pull/27252#discussion_r1228356872
+        # temporary solution, testing access for all streams to avoid 402 error
+        streams = [Annotations, Cohorts, Engage, Export, Revenue]
+        for stream_class in streams:
+            try:
+                stream = stream_class(authenticator=auth, **config)
+                stream.reqs_per_hour_limit = 0
```

Comment:
Why is the stream’s limit set to 0?

Reply:
Thanks, this is useless here, also I realized  that in case when all streams have 402 error this method returns true, so I added logic to return false if all streams with 402 error